File: array/two_sum/two_sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	dic := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		num := nums[i]
		if idx, ok := dic[num]; ok {
			return []int{idx, i}
		}
		dic[target-num] = i
	}
	return []int{}
}

func main() {
	arr := []int{3, 2, 4}
	target := 6
	res := twoSum(arr, target)
	fmt.Println(res)
}

File: array/two_sum/two_sum.rb
def two_sum(nums, target)
  dic = {}
  nums.each.with_index do |num, i|
    return [dic[num], i] if dic.include? num
    dic[target - num] = i
  end
end

arr = [3, 2, 4]
target = 6
res = two_sum(arr, target)
p res
